packages/core/src/provider/base.ts

```diff
@@ -181,16 +191,18 @@ export class SmartAccountProvider<
   waitForUserOperationTransaction = async (hash: Hash): Promise<Hash> => {
     for (let i = 0; i < this.txMaxRetries; i++) {
       await new Promise((resolve) =>
-        setTimeout(resolve, this.txRetryIntervalMs)
+        setTimeout(
+          resolve,
+          this.txRetryIntervalMs * Math.pow(this.txRetryMulitplier, i)
```

Comment:
nit: usually for exponential retries, there's also random jitter added to the final value here -- usually a random number in the hundreds of milliseconds

logic is that if the reason these retries are happening because we went down, we don't want our clients to keep us down by causing a thundering herd scenario. So you randomly stagger each of the clients to try to avoid all of them making calls at the same time